File: Prototype-API-test-Java/app/src/main/java/tn743/ufrrj/gcampus/j_g_campus_test/gui/acti/MainActivity.java
package tn743.ufrrj.gcampus.j_g_campus_test.gui.acti;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import tn743.ufrrj.gcampus.j_g_campus_test.R;
import tn743.ufrrj.gcampus.j_g_campus_test.logic.Configure;
import tn743.ufrrj.gcampus.j_g_campus_test.logic.GameManagerGCampus;
import tn743.ufrrj.gcampus.j_g_campus_test.menumain.Function;
import tn743.ufrrj.gcampus.j_g_campus_test.menumain.FunctionsAdapter;


public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";
    public static final int SETUP = 0;
    public static final int CHALLENGES = 1;
    public static final int CHALLENGES_IN_PLAYING = 2;
    public static final int SCORE = 3;
    public static final int GPSCHALLENGES = 4;
    public static final int QRCODE_R = 5;
    public static final int QRCODE_W = 6;



    /*
    private lateinit var mRVFunctions: RecyclerView
    private lateinit var mMessageReceiver: MyBroadcastReceive
    private val TAG = "MainActivity"
 */
    private RecyclerView mRVFunctions = null;
    private MyBroadcastReceive mMessageReceiver = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mRVFunctions = findViewById(R.id.rvFunctions);
        ArrayList<Function> functions = Function.createFunctionList();
        FunctionsAdapter adapter = new FunctionsAdapter(functions);
        mRVFunctions.setAdapter(adapter);
        mRVFunctions.setLayoutManager(new LinearLayoutManager(this));
        mMessageReceiver = new MyBroadcastReceive();
        LocalBroadcastManager.getInstance(this).registerReceiver(mMessageReceiver, new IntentFilter("selected-test"));

        Configure.getInstance().setContext(getApplicationContext());
        GameManagerGCampus.getInstance().init(getApplicationContext());
        GameManagerGCampus.getInstance().loadChallengesInProgress();

    }//protected void onCreate(Bundle savedInstanceState) {
    @Override
    protected void onDestroy() {
        // Unregister since the activity is about to be closed.
        LocalBroadcastManager.getInstance(this).unregisterReceiver(mMessageReceiver);

        super.onDestroy();
    }

    private void nextActivity(int opt){
        Intent indent = null;
        switch (opt){
            case SETUP :
                indent = new Intent(this, ConfigActivity.class);
                break;
            case CHALLENGES :
                indent = new Intent(this, ChallengesActivity.class);
                indent.putExtra("type", false);
                break;
            case CHALLENGES_IN_PLAYING:
                indent = new Intent(this, ChallengesActivity.class);
                indent.putExtra("type", true);
                break;
            case SCORE :
                break;
            case GPSCHALLENGES :
                break;
            case QRCODE_R :
            case QRCODE_W :
                indent = new Intent(this, QRCodeActivity.class);
                indent.putExtra("type", opt);
                break;

        }
        if (indent != null)
            startActivity(indent);

    }//private void nextActivity(int opt){
    private class MyBroadcastReceive extends BroadcastReceiver {

        @Override
        public void onReceive(Context context, Intent intent) {
            if (intent != null){
                int opt = intent.getIntExtra("parameter", -1);
                if (opt != -1){
                    nextActivity(opt);
                }//if (opt != -1){
            }//if (intent != null){

        }//public void onReceive(Context context, Intent intent) {
    }//private class MyBroadcastReceive extends BroadcastReceiver {
}//public class MainActivity extends AppCompatActivity {

File: Prototipo-API/app/src/main/java/tn743/ufrrj/gcampus/test/MainActivity.kt
package tn743.ufrrj.gcampus.test

import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.util.DisplayMetrics
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView


class MainActivity : AppCompatActivity() {
    private lateinit var mRVFunctions: RecyclerView
    private lateinit var mMessageReceiver: MyBroadcastReceive
    private val TAG = "MainActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
        //enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        //mRVFunctions = this.findViewById (R.id.rvFunctions)
        mRVFunctions = findViewById(R.id.rvFunctions)
        var funcs = Function.createFunctionList()
        var adapter:FunctionsAdapter = FunctionsAdapter(funcs, applicationContext)
        mRVFunctions.adapter = adapter
        mRVFunctions.layoutManager = LinearLayoutManager(this)

        mMessageReceiver = MyBroadcastReceive()
        LocalBroadcastManager.getInstance(this)
            .registerReceiver(mMessageReceiver, IntentFilter("selected-test"))

    }//override fun onCreate(savedInstanceState: Bundle?) {

    private fun nextActivity(index:Int){
        var width:Int = 0
        var height:Int = 0

        var msg:String = "(" + width.toString() + "," + height.toString() + ")";
        Log.d("TAG", msg)
        val activity = applicationContext as Activity
        val displayMetrics = DisplayMetrics()
        val windowManager = activity.getWindowManager()
        //getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
        val s = windowManager.currentWindowMetrics.toString()
        Log.d(TAG, s)


        //var next:Intent = Intent(this, LogActivity::class.java)
        //next.putExtra("type", index)
        //startActivity(next)

    }

    inner private class MyBroadcastReceive: BroadcastReceiver(){
        override fun onReceive(context: Context?, intent: Intent?) {
            intent?.let {
                var index:Int = it.getIntExtra("parameter", -1)
                if (index == -1) {
                    Toast.makeText(applicationContext, "Opção inválida!", Toast.LENGTH_LONG).show();
                    return;
                }
                nextActivity(index);
            }



        }//override fun onReceive(context: Context?, intent: Intent?) {

    }//inner private class MyBroadcastReceive: BroadcastReceiver(){

}//class MainActivity : AppCompatActivity() {